### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by vladlens-bogdanovs-sub-zero**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Homework task description
+
+Implement small application which will be able to handle payment import CSV files (batches) and incoming API calls (
+single payment).
+
+Data to work with can be found in `exampleData/` folder.
+Payment import file and API call example can be found in `import/` folder.
```
Comment (fruit) [inline readme.md]: > "and API call example"

there're no such API call examples in folder `import/`

File: readme.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+Payment import and request has loan number in reference and|or in description. Consists of 2 letters and 8 numbers,
```
Comment (fruit) [inline readme.md]: > Payment import and request

something wrong here, API request?

File: readme.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+### Technical requirements
+
+1. PHP;
+2. Data storage could be anything: files, SQLite, PostgreSQL, etc.
+3. Use only libraries and components that are necessary. No need to use full framework.
+4. Two entry points to import payment(-s)
+    1. API (single payment)
+    2. Console (CSV payment batch)
+5. Different error handling
+   - API: Duplicate entry - 409, rest of errors - 400, All fine - 2XX depending on implementation
```
Comment (fruit) [inline readme.md]: Invalid JSON payload `422`

**Review by fruit — CHANGES_REQUESTED** [review]
- In technical requirements I would add how should look API endpoint (at which path)
- somewhere we lose, how should look like payment import script
- also, we could ask to add graceful CSV payment import termination (i.e. when someone presses <kbd>CTRL + C</kbd>)
- rename readme.md to README.md

File: readme.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+Validation before creation
+   - duplicate entry (paymentReference or refId)
+   - negative amount
+   - invalid date
+
+Save to the storage
+
+#### Payment assignment
+
+Payment amount === loan.amount_to_pay
```
Comment (fruit) [inline readme.md]:
```suggestion
When payment amount equal to matched loan's amount to pay:
```

File: readme.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+   - Mark loan as paid
+   - Mark payment as assigned
+
+Payment amount > loan.amount_to_pay
```
Comment (fruit) [inline readme.md]:
```suggestion
When payment amount greater than loan's amount to pay:
```

File: readme.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+   - Mark loan as paid
+   - Mark payment as partially assigned
+   - Create refund payment as separate entity called "Payment Order" with all necessary information
+
+Payment amount < loan.amount_to_pay
```
Comment (fruit) [inline readme.md]:
```suggestion
When payment amount is less than loan's amount to pay:
```

**Review by fruit — CHANGES_REQUESTED** [review]

File: readme.md (hunk 3 above)
Comment (vladlens-bogdanovs-sub-zero) [reply readme.md]: left 400

File: readme.md (hunk 1 above)
Comment (vladlens-bogdanovs-sub-zero) [reply readme.md]: moved everything into exampleData

File: README.md [hunk 7]
```diff
@@ -1,2 +1,67 @@
-# docs-home-task-nordic
-Nordic backend developer home task.
+# Homework task description
+
+Implement small application which will be able to handle payment import CSV files (batches) and incoming API calls (
+single payment).
```
Comment (fruit) [inline README.md]:
```suggestion
Implement small application which will be able to handle payment import CSV files (batches) and incoming API calls (single payment).
```

File: README.md [hunk 8]
```diff
@@ -1,2 +1,66 @@
...
+
+Track paid amounts on loan
+
+#### Console interface
+- Show payments by date `report --date=YYYY-MM-DD`
+
+#### Communication functionality implementation (abstraction layer is enough)
+- Payment received: communication sent to email and|or phone if defined
+- Loan fully paid: communication sent to email and|or phone if defined
+- Failed payments report: support@email.com
```
Comment (fruit) [inline README.md]:
```suggestion
- Failed payments report: support@example.com
```

**Review by fruit — APPROVED** [review]

**Pull request merged**